app/pages/my-dashboard/integrations.tsx

```diff
@@ -1,21 +1,84 @@
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import IntegrationInfoTabs from 'page-partials/my-dashboard/IntegrationInfoTabs';
 import IntegrationList from 'page-partials/my-dashboard/IntegrationList';
 import VerticalLayout from 'page-partials/my-dashboard/VerticalLayout';
 import { Integration } from 'interfaces/Request';
 import { PageProps } from 'interfaces/props';
+import CenteredModal from '@app/components/CenteredModal';
+import { useRouter } from 'next/router';
+import { faCommentDots, faEnvelope, faExclamationTriangle } from '@fortawesome/free-solid-svg-icons';
+import styled from 'styled-components';
+import { Link } from '@button-inc/bcgov-theme';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+const Column = styled.div`
+  display: flex;
+  flex-direction: column;
+  flex-basis: 100%;
+  flex: 1;
+`;
 
 function MyIntegrations({ session }: PageProps) {
+  const router = useRouter();
   const [integration, setIntegration] = useState<Integration | null>(null);
   const [integrationCount, setIntegrationCount] = useState(1);
+  const [showModal, setShowModal] = useState(false);
+  const integrationFailedMessageModalId = 'integration-failed-modal';
+  const handleModalFailedMessageModal = async () => (window.location.hash = integrationFailedMessageModalId);
+
+  useEffect(() => {
+    // Check if the query parameter 'showModal' is present and set to 'true'
+    if (router.query.integrationFailedMessageModal === 'true') {
```

Comment:
will the query always have an integrationFaildMessageModal field? or should this be `query?.`

Reply:
yes that's a good suggestion Jon, added it